### PR TITLE
The solution for progress bar

### PR DESCRIPTION
**Pull request by quynhn**

### CODE REVIEW

File: jame3-progressing/index.html [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+      });
+    }
+
+    function getData(callback) {
+      var req = new XMLHttpRequest();
+      req.addEventListener('load', function() {
+        callback(JSON.parse(this.responseText))
+      });
+      req.open('GET', 'http://rawgit.com/quynhn/james-project/jame3-progressing/jame3-progressing/progress.json');
```
Comment (mbaechler) [inline jame3-progressing/index.html]: it should take the json relatively to the html file

File: jame3-progressing/progress.json [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+     "name" : "make sure default (JPA + ES) backends work well",
+     "size" : 22
+   },
+   {
+     "name" : "make a security audit",
+     "size" : 23
+   },
+   {
+     "name" : "bug triaging",
+     "size" : 24
```
Comment (mbaechler) [inline jame3-progressing/progress.json]: could we change that to "completion"  ?

File: jame3-progressing/index.html (hunk 1 above)
Comment (quynhn) [inline jame3-progressing/index.html]: Done

File: jame3-progressing/progress.json (hunk 2 above)
Comment (quynhn) [inline jame3-progressing/progress.json]: Done

**Comment by quynhn** [comment]
`1. an progress bar that is the average of others in each graph
    2. changing color from red (0%) to blue (100%) with completion.
    3. adding a complexity factor on tasks`
1 --> Done
2 --> I tried to find the way but still don't know how to :(
3 --> sorry forgot to ask you but how can I calculate the complexity of task?

File: jame3-progressing/progress.json~ [hunk 3]
```diff
@@ -0,0 +1,100 @@
+{
```
Comment (chibenwa) [inline jame3-progressing/progress.json~]: Is this file required ?

File: jame3-progressing/progress.json [hunk 4]
```diff
@@ -0,0 +1,100 @@
+{
```
Comment (chibenwa) [inline jame3-progressing/progress.json]: Does completion data have to be accurate here ?

File: jame3-progressing/progress.json~ (hunk 3 above)
Comment (quynhn) [inline jame3-progressing/progress.json~]: It's noisy :(. Deleted

File: jame3-progressing/progress.json (hunk 4 above)
Comment (quynhn) [inline jame3-progressing/progress.json]: Do you mean the completion value can be float data? Could you explain me more?

**Comment by quynhn** [comment]
Done for "Adding a complexity factor on tasks"

File: jame3-progressing/progress.json (hunk 4 above)
Comment (chibenwa) [inline jame3-progressing/progress.json]: No, I mean, did you put random input or do you need it to be real data reflecting actual task completion ?

Comment (quynhn) [inline jame3-progressing/progress.json]: yes, it's random input for now so whenever the real data is changed, it needs to change json file manually. That is the requirement :)

**Comment by mbaechler** [comment]
screenshot from 2016-10-03 09-04-34

The global progress should be per graph, not a separate graph.
I don't think we should display complexity as a bar, it's confusing, maybe just put it in the tooltip ?
And finally, about the color, don't bother if it's not trivial.

**Comment by quynhn** [comment]
Done the comment

**Comment by mbaechler** [comment]
Very cool. Could I ask for a last modification : I'd like the x scale to always be 100.

**Comment by quynhn** [comment]
Done

**Comment by aduprat** [comment]
test this please

**Comment by mbaechler** [comment]
squashed here : https://github.com/linagora/james-project/pull/445
